homeassistant/components/rituals_perfume_genie/switch.py

```diff
@@ -0,0 +1,92 @@
+"""Support for Rituals Perfume Genie switches."""
+from datetime import timedelta
+
+from homeassistant.components.switch import SwitchEntity
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ON_STATE = "1"
+AVAILABLE_STATE = 1
+
+MANUFACTURER = "Rituals Cosmetics"
+MODEL = "Diffuser"
+ICON = "mdi:fan"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the diffuser switch."""
+    account = hass.data[DOMAIN][config_entry.entry_id]
+    diffusers = await account.get_devices()
+
+    entities = []
+    for diffuser in diffusers:
+        entities.append(DiffuserSwitch(diffuser))
+
+    async_add_entities(entities, True)
+
+
+class DiffuserSwitch(SwitchEntity):
+    """Representation of a diffuser switch."""
+
+    def __init__(self, diffuser):
+        """Initialize the switch."""
+        self._diffuser = diffuser
+
```

Comment:
If it makes sense in this case:

```
    @property
    def _hub(self):
        """Return hub data."""
        return self._diffuser.data["hub"]
```

Then you can replace `self._diffuser.data["hub"]` with `self._hub` everywhere.

Reply:
I don't see how this makes the code more efficient or clearer. It's indeed a little shorter but don't think that adds much value.